File: flutter_client/lib/native_contract/auth/request_email_change_request_dto.dart
class RequestEmailChangeRequestDto {
  const RequestEmailChangeRequestDto({
    required this.newEmail,
    required this.currentPassword,
  });

  final String newEmail;
  final String currentPassword;

  Map<String, dynamic> toJson() => {
    'new_email': newEmail,
    'current_password': currentPassword,
  };
}

File: cloud_backend/src/main/java/com/excellentcalendar/cloud/identity/api/RequestEmailChangeRequestDto.java
package com.excellentcalendar.cloud.identity.api;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record RequestEmailChangeRequestDto(
        @NotBlank @Email @Size(max = 254) String newEmail,
        @NotBlank @Size(min = 1, max = 128) String currentPassword) {
}
